src/app/filelogger.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2016  sledgehammer999 <hammered999@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include <QDir>
+#include <QFile>
+#include <QTextStream>
+#include "filelogger.h"
+#include "base/preferences.h"
+#include "base/logger.h"
+#include "base/utils/fs.h"
+
+FileLogger::FileLogger()
+{
+    configure();
+    const Logger* const logger = Logger::instance();
+    foreach (const Log::Msg& msg, logger->getMessages())
+        addLogMessage(msg);
+
+    connect(Preferences::instance(), SIGNAL(changed()), SLOT(configure()));
+    connect(logger, SIGNAL(newLogMessage(const Log::Msg &)), SLOT(addLogMessage(const Log::Msg &)));
+}
+
+void FileLogger::configure()
+{
+    const Preferences* const pref = Preferences::instance();
+    m_path = Utils::Fs::fromNativePath(pref->fileLogPath());
+    if (!m_path.endsWith('/'))
+        m_path += "/";
+    QDir dir(m_path);
+    if (!dir.exists())
+        dir.mkpath(m_path);
+    m_path += "qbittorrent.log";
+    m_manage = pref->fileLogManaged();
+    m_size = pref->fileLogMaxSize();
+    m_fileLogAction = static_cast<FileLogAction>(pref->fileLogAction());
+}
+
+void FileLogger::addLogMessage(const Log::Msg &msg)
+{
+    QFile log(m_path);
+    if (!log.open(QIODevice::WriteOnly | QIODevice::Append | QIODevice::Text))
+        return;
```

Comment:
It makes sense (we can safely assume that if we are unable to open the file once, we will not be able to do it next). Although it is better to name the flag something more suitable (`m_invalid` or something similar).